### PR TITLE
Adds dedicated sprites for metal hydrogen armor

### PR DESCRIPTION
**Pull request by dawsonkeyes**
## About The Pull Request

Adds two new (nicely made) sprite sets for the metal hydrogen armor and helmet, which previously reused bad-looking sprites and thus blended terribly.

## Why It's Good For The Game

No more butt-fugly sprite reuses!
image

## Changelog
:cl:
add: New metal hydrogen armor sprites
/:cl:

### CODE REVIEW

**Comment by dawsonkeyes** [comment]
why is this marked as feature

**Comment by Tennessee116** [comment]
I'm no spriter but that armor looks like pillows.

**Comment by necromanceranne** [comment]
> why is this marked as feature

You used the add and not balance tag

**Comment by dawsonkeyes** [comment]
> > why is this marked as feature
> 
> You used the add and not balance tag

gotcha

File: code/modules/clothing/head/helmet.dm [hunk 1]
```diff
@@ -374,9 +374,9 @@
 /obj/item/clothing/head/helmet/elder_atmosian
 	name = "Elder Atmosian Helmet"
```
Comment (necromanceranne) [inline code/modules/clothing/head/helmet.dm:375]:
```suggestion
	name = "/improper Elder Atmosian Helmet"
```

File: code/modules/clothing/suits/armor.dm [hunk 2]
```diff
@@ -297,10 +297,10 @@
 /obj/item/clothing/suit/armor/elder_atmosian
 	name = "Elder Atmosian Armor"
```
Comment (necromanceranne) [inline code/modules/clothing/suits/armor.dm:298]:
```suggestion
	name = "/improper Elder Atmosian Armor"
```

File: code/modules/clothing/head/helmet.dm (hunk 1 above)
Comment (necromanceranne) [reply code/modules/clothing/head/helmet.dm]: This is mostly so you don't get stuff like 

'Urist McGreyshit throws Elder Atmosian Helmet'

And instead displays as

'Urist McGreyshit throws the Elder Atmosian Helmet'

**Comment by RaveRadbury** [comment]
Atmos power creep

**Comment by tralezab** [comment]
atomize 😡

**Comment by Qustinnus** [comment]
this is not "metal hydrogen" armor, it's material armor and it affects all the material armours that exist. show a comparision picture with of all types of materials applied to the armor before and after.

**Comment by Qustinnus** [comment]
post before / after pictures, else this is not very useful.

File: code/modules/clothing/head/helmet.dm [hunk 3]
```diff
@@ -372,11 +372,11 @@
...
 /obj/item/clothing/head/helmet/elder_atmosian
-	name = "Elder Atmosian Helmet"
+	name = "/improper Elder Atmosian Helmet"
 	desc = "A superb helmet made with the toughest and rarest materials available to man."
-	icon_state = "knight_greyscale"
-	inhand_icon_state = "knight_greyscale"
-	armor = list(MELEE = 15, BULLET = 10, LASER = 30, ENERGY = 30, BOMB = 10, BIO = 10, RAD = 20, FIRE = 65, ACID = 40, WOUND = 15)
+	icon_state = "h2helmet"
+	inhand_icon_state = "h2helmet"
+	armor = list(MELEE = 50, BULLET = 30, LASER = 40, ENERGY = 40, BOMB = 20, BIO = 10, RAD = 40, FIRE = 100, ACID = 40, WOUND = 30)
```
Comment (Qustinnus) [inline code/modules/clothing/head/helmet.dm:379]: No, the armour gets its values from the material. Don't buff it like this.

Comment (necromanceranne) [reply code/modules/clothing/head/helmet.dm:379]: To explain this more accurately, materials apply a multiplier to the armor values. If you want it to use specifically these values, remove the MATERIAL_AFFECT_STATISTICS flag from the material_flags.

File: code/modules/clothing/head/helmet.dm [hunk 4]
```diff
@@ -372,11 +372,11 @@
...
 	material_flags = MATERIAL_COLOR | MATERIAL_AFFECT_STATISTICS //Can change color and add prefix
```
Comment (Ghilker) [inline code/modules/clothing/head/helmet.dm:380]: Hey thanks for the work! Please remove these material_flags from here and from the armor, and add custom_materials to both the armor and the helmet

**Comment by Ghilker** [comment]
the sprites looks dope!

File: code/modules/clothing/head/helmet.dm (hunk 4 above)
Comment (Ghilker) [reply code/modules/clothing/head/helmet.dm:380]: so something like `custom_materials = list(/datum/material/metalhydrogen = 5)`, the amount is the same of the number of sheets used in crafting (5 for helmet, 8 for armor)

**Comment by dawsonkeyes** [comment]
> also this makes the armor massively OP,

I agree that it shouldn’t be morr powerful than riot gear, the values have been made more reasonable

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Put the balance changes in their own PR. Or don't do them at all. I don't think we should be incentivizing the single player atmos tech gameplay.

File: code/modules/clothing/head/helmet.dm [hunk 5]
```diff
@@ -372,14 +372,16 @@
 	strip_delay = 80
 
 /obj/item/clothing/head/helmet/elder_atmosian
-	name = "Elder Atmosian Helmet"
+	name = "/improper Elder Atmosian Helmet"
```
Comment (Mothblocks) [inline code/modules/clothing/head/helmet.dm:375]: You didn't test this.

```suggestion
	name = "\improper Elder Atmosian Helmet"
```

Comment (dawsonkeyes) [reply code/modules/clothing/head/helmet.dm:375]: forward slash, backslash, woe is me

**Comment by dawsonkeyes** [comment]
> Put the balance changes in their own PR. Or don't do them at all. I don't think we should be incentivizing the single player atmos tech gameplay.

What about this incentivizes single player atmos? Attaining the required gasses at the right concentrations can be a slog solo, let alone getting a high-quality burn capable of heating it to the right temperature- it would take far longer to attain such armor solo than it would cooperating with other techies or engineers.

**Comment by Mothblocks** [comment]
An atmos tech making metal hydrogen is completely irrelevant to the crew as a whole.

Atomization is required, and not my opinion.

File: code/modules/clothing/head/helmet.dm [hunk 6]
```diff
@@ -372,14 +372,16 @@
...
+	name = "\improper Elder Atmosian Helmet"
 	desc = "A superb helmet made with the toughest and rarest materials available to man."
-	icon_state = "knight_greyscale"
-	inhand_icon_state = "knight_greyscale"
-	armor = list(MELEE = 15, BULLET = 10, LASER = 30, ENERGY = 30, BOMB = 10, BIO = 10, RAD = 20, FIRE = 65, ACID = 40, WOUND = 15)
-	material_flags = MATERIAL_COLOR | MATERIAL_AFFECT_STATISTICS //Can change color and add prefix
+	icon_state = "h2helmet"
+	inhand_icon_state = "h2helmet"
+	armor = list(MELEE = 40, BULLET = 20, LASER = 30, ENERGY = 30, BOMB = 20, BIO = 10, RAD = 40, FIRE = 100, ACID = 40, WOUND = 30)
+	custom_materials = list(/datum/material/metalhydrogen = 5)
```
Comment (Fikou) [inline code/modules/clothing/head/helmet.dm:380]: 5??

File: code/modules/clothing/suits/armor.dm [hunk 7]
```diff
@@ -295,15 +295,16 @@
...
+	name = "\improper Elder Atmosian Armor"
 	desc = "A superb armor made with the toughest and rarest materials available to man."
-	icon_state = "knight_grey"
-	inhand_icon_state = "knight_greyscale"
-	material_flags = MATERIAL_COLOR | MATERIAL_AFFECT_STATISTICS//Can change color and add prefix
-	armor = list(MELEE = 15, BULLET = 10, LASER = 30, ENERGY = 30, BOMB = 10, BIO = 10, RAD = 20, FIRE = 65, ACID = 40, WOUND = 15)
+	icon_state = "h2armor"
+	inhand_icon_state = "h2armor"
+	custom_materials = list(/datum/material/metalhydrogen = 8)
+	armor = list(MELEE = 45, BULLET = 20, LASER = 30, ENERGY = 30, BOMB = 20, BIO = 10, RAD = 40, FIRE = 100, ACID = 100, WOUND = 30)
```
Comment (Fikou) [inline code/modules/clothing/suits/armor.dm:303]: inconsistent armor numbers with the helmet

File: code/modules/clothing/suits/armor.dm [hunk 8]
```diff
@@ -295,15 +295,16 @@
...
-	name = "Elder Atmosian Armor"
+	name = "\improper Elder Atmosian Armor"
 	desc = "A superb armor made with the toughest and rarest materials available to man."
-	icon_state = "knight_grey"
-	inhand_icon_state = "knight_greyscale"
-	material_flags = MATERIAL_COLOR | MATERIAL_AFFECT_STATISTICS//Can change color and add prefix
-	armor = list(MELEE = 15, BULLET = 10, LASER = 30, ENERGY = 30, BOMB = 10, BIO = 10, RAD = 20, FIRE = 65, ACID = 40, WOUND = 15)
+	icon_state = "h2armor"
+	inhand_icon_state = "h2armor"
+	custom_materials = list(/datum/material/metalhydrogen = 8)
```
Comment (Fikou) [inline code/modules/clothing/suits/armor.dm:302]: 8?

File: code/modules/clothing/suits/armor.dm [hunk 9]
```diff
@@ -295,15 +295,16 @@
...
+	armor = list(MELEE = 45, BULLET = 20, LASER = 30, ENERGY = 30, BOMB = 20, BIO = 10, RAD = 40, FIRE = 100, ACID = 100, WOUND = 30)
 	body_parts_covered = CHEST|GROIN|LEGS|FEET|ARMS|HANDS
 	cold_protection = CHEST|GROIN|LEGS|FEET|ARMS|HANDS
 	heat_protection = CHEST|GROIN|LEGS|FEET|ARMS|HANDS
```
Comment (Fikou) [inline code/modules/clothing/suits/armor.dm:306]: the sprites clearly show it not covering the arms, hands and feet

File: code/modules/clothing/head/helmet.dm (hunk 6 above)
Comment (dawsonkeyes) [reply code/modules/clothing/head/helmet.dm:380]: 5 and 8 are the amounts of material used to make each armor piece

Comment (Qustinnus) [reply code/modules/clothing/head/helmet.dm:380]: a sheet is 2000 units of material

Comment (dawsonkeyes) [reply code/modules/clothing/head/helmet.dm:380]: my brain is not massive

File: code/modules/clothing/head/helmet.dm [hunk 10]
```diff
@@ -372,12 +372,12 @@
...
-	name = "Elder Atmosian Helmet"
+	name = "\improper Elder Atmosian Helmet"
 	desc = "A superb helmet made with the toughest and rarest materials available to man."
-	icon_state = "knight_greyscale"
-	inhand_icon_state = "knight_greyscale"
+	icon_state = "h2helmet"
+	inhand_icon_state = "h2helmet"
 	armor = list(MELEE = 15, BULLET = 10, LASER = 30, ENERGY = 30, BOMB = 10, BIO = 10, RAD = 20, FIRE = 65, ACID = 40, WOUND = 15)
-	material_flags = MATERIAL_COLOR | MATERIAL_AFFECT_STATISTICS //Can change color and add prefix
+	custom_materials = list(/datum/material/metalhydrogen = 10000)
```
Comment (Mothblocks) [inline code/modules/clothing/head/helmet.dm:380]: This is still a behavior change, move this out to a different PR.

This shouldn't really touch anything other than sprites (and thus, icon states).

File: code/modules/clothing/suits/armor.dm [hunk 11]
```diff
@@ -295,11 +295,11 @@
...
 /obj/item/clothing/suit/armor/elder_atmosian
-	name = "Elder Atmosian Armor"
+	name = "\improper Elder Atmosian Armor"
 	desc = "A superb armor made with the toughest and rarest materials available to man."
-	icon_state = "knight_grey"
-	inhand_icon_state = "knight_greyscale"
-	material_flags = MATERIAL_COLOR | MATERIAL_AFFECT_STATISTICS//Can change color and add prefix
+	icon_state = "h2armor"
+	inhand_icon_state = "h2armor"
+	custom_materials = list(/datum/material/metalhydrogen = 16000)
```
Comment (Mothblocks) [inline code/modules/clothing/suits/armor.dm:302]: This too--keep it to sprites.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Review by Mothblocks — APPROVED** [review]
Code is good, up to Twaticus now.
